### ISSUE TITLE
Can I have a local menubar for each window?

### ISSUE THREAD

**Issue by geppi** [isl-org/Open3D#5629]
### Checklist

- [X] I have searched for [similar issues](https://github.com/isl-org/Open3D/issues).
- [X] For Python issues, I have tested with the [latest development wheel](http://www.open3d.org/docs/latest/getting_started.html#development-version-pip).
- [X] I have checked the [release documentation](http://www.open3d.org/docs/release/) and the [latest documentation](http://www.open3d.org/docs/latest/) (for `master` branch).

### My Question

The current gui design only has a global menubar for all application windows that is connected to the gui.Application.instance.
Checkable items like "Show menu" in the "Settings" menu entry of the example applications do therefore not necessarily reflect the state of the curreent window. You can e.g. hide the settings panel in one window while it is still visible in others. However, the "Show menu" item is displayed unchecked in all windows because of the global nature of the menubar.

**Comment by decamargo10** [comment]
Agreed, there should be an option to change the menu for a new window. It would help a lot already if there would be a way to check from which window a menu call is coming from. But as far as I know, there is no way to do that yet. If there is I would be more than happy to learn about it-
